examples/jsm/math/MeshSurfaceSampler.js

```diff
@@ -200,6 +224,29 @@ class MeshSurfaceSampler {
 			targetColor.g = _color.y;
 			targetColor.b = _color.z;
 
+		} else if ( targetColor !== undefined && this.uvAttribute !== undefined ) {
+
+			_uv.set( 0, 0, 0 ).addScaledVector( _face.a, u ).addScaledVector( _face.b, v ).addScaledVector( _face.c, 1 - ( u + v ) );
+
+			_uv1.fromBufferAttribute( this.uvAttribute, faceIndex * 3 );
+
+			_uv2.fromBufferAttribute( this.uvAttribute, faceIndex * 3 + 1 );
+
+			_uv3.fromBufferAttribute( this.uvAttribute, faceIndex * 3 + 2 );
+
+			_face.getUV( _uv, _uv1, _uv2, _uv3, targetUV );
+
+			u = targetUV.x;
+			v = targetUV.y;
+
+			const tx = Math.min( u * this.colorMapData.width | 0, this.colorMapData.width - 1 );
+			const ty = Math.min( v * this.colorMapData.height | 0, this.colorMapData.height - 1 );
+			const offset = ( ty * this.colorMapData.width + tx ) * 4;
+
+			targetColor.r = this.colorMapData.data[ offset ] / 255;
+			targetColor.g = this.colorMapData.data[ offset + 1 ] / 255;
+			targetColor.b = this.colorMapData.data[ offset + 2 ] / 255;
```

Comment:
Related to the previous comment I think these lines should be remove and it should be the users responsibility sample any 2d data outside of this function. Additionally not all color data is available on the CPU and texture sampling interpolation can be a complicated / use-case specific topic.

Reply:
This original demo is an instancing demo but it is also the linked demo for meshsurfacesampler. Original demo just had a color palette for randomly selecting a color, and the targetColor returned from meshsurfacesampler was undefined. Ideally there is a meshsurfacesampler demo to demonstrate returning of targetColor for the sampled point. returning UV target coordinate is also useful for the project I'm working on.